public/offline.html

```diff
@@ -174,38 +187,74 @@ <h4>(pick a color below && start drawing!)</h4>
     </g>
   </svg>
   <script type="text/javascript">
+
+    const palettes = {
+      default: ["#F4908E", "#F2F097", "#88B0DC", "#F7B5D1", "#53C4AF", "#FDE38C"],
+      // Color-deficient accessible palettes: https://owi.usgs.gov/blog/tolcolors/
+      paulTol: ["#B997C6", "#824D99", "#4E79C4", "#57A2AC", "#7EB875", "#D0B440", "#E67F33"],
+      // Add more palette options below and to the "select" dropdown.
+    }
+    let activePalette = 'default' // match key inside the palettes variable
+
+    // Globally shared DOM nodes
     const canvas = document.querySelector('canvas')
     const context = canvas.getContext('2d')
-    const colors = ["#F4908E", "#F2F097", "#88B0DC", "#F7B5D1", "#53C4AF", "#FDE38C"]
     const colorDiv = document.querySelector(".colors")
 
-    colors.forEach(color => {
-      const button = document.createElement("button")
-      button.classList.add("color")
-      button.style.backgroundColor = color
-      colorDiv.appendChild(button)
-      button.addEventListener('click', () => context.strokeStyle = color)
-    })
+    // handler for color input buttons
+    function handleButtonClick(event) {
+      const buttonStyle = event.target.style
+      context.strokeStyle = buttonStyle.backgroundColor
+    }
+
+    // Main drawing function
+    function renderColorButtons(colors) {
+      // First, remove existing event listeners to prevent memory leak
+      colorDiv.querySelectorAll('button').forEach((button) => {
+        button.removeEventListener('click', handleButtonClick)
+      })
+      // Then, remove existing buttons
+      colorDiv.innerHTML = ""
```

Comment:
There was some SO discussion about the best way to do this, I think this solution is easy to ready + performance across browsers is not a big concern when we're talking about small (less than 10) elements.

https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript